todo_app/data/item.py

```diff
@@ -0,0 +1,142 @@
+- TRELLO_COMPLETE_LIST_ID: The ID of the done list on the Trello board
+"""
+
+import os
+
+
+TRELLO_NS_LIST_ID = os.getenv("TRELLO_NS_LIST_ID")
+TRELLO_COMPLETE_LIST_ID = os.getenv("TRELLO_COMPLETE_LIST_ID")
+
+
+class Item:
+    def __init__(self, title, id=None, id_list=TRELLO_NS_LIST_ID,
+                 description=None, due_date=None, status='Not Started'):
+        """Initialize a new Item with the given ID, title, and status."""
+        self.title = title
+        self.id = id
+        self.id_list = id_list
+        self.description = description
+        self.due_date = due_date
+        self.status = status
+
+    @classmethod
+    def translate_trello_card_to_item(cls, trello_card):
+        """
+        Translates a Trello card to an item dictionary as per the old
+        structure.
+
+        Args:
+            trello_card (dict): The Trello card data.
+
+        Returns:
+            dict: The translated item.
+        """
+
+        # Extract required fields from the Trello card
+        title = trello_card.get('name')
+        id = trello_card.get('id')
+        id_list = trello_card.get('idList')
+        description = trello_card.get('desc')
+        due_date = trello_card.get('due')
+
+        # Determine status based on the list ID
+        status = 'Not Started' if id_list == TRELLO_NS_LIST_ID else 'Complete'
+
+        return cls(title, id, id_list, description, due_date, status)
+
+    def get_id(self):
+        """
+        Returns the id of the item.
+
+        Returns:
+            str: The id of the item.
+        """
+        return self.id
+
+    def get_id_list(self):
+        """
+        Returns the list id that the item belongs to.
+
+        Returns:
+            str: The id of the item's list.
+        """
+        return self.id_list
+
+    def get_description(self):
+        """
+        Returns the item description.
+
+        Returns:
+            str: The description of the item.
+        """
+        return self.description
+
+    def get_due_date(self):
+        """
+        Returns the item due date.
+
+        Returns:
+            str: The due date of the item.
+        """
+        return self.due_date
+
+    def get_title(self):
+        """
+        Returns the title of the item.
+
+        Returns:
+            str: The title of the item.
+        """
+        return self.title
+
+    def get_status(self):
+        """
+        Returns the status of the item.
+
+        Returns:
+            str: The status of the item.
+        """
+        return self.status
+
+    def is_complete(self):
+        """Check if the item is marked as complete.
+
+        Returns:
+            bool: True if the item is complete, False otherwise.
+        """
+        return self.status == 'Complete'
+
+    def mark_as_complete(self):
+        """Mark the item as complete."""
+        self.status = 'Complete'
+        self.id_list = TRELLO_COMPLETE_LIST_ID
```

Comment:
Given that these properties _should_ always move in tandem, would it make sense to actually only track one and then just derive the other directly from that single source of truth? E.g. if we tracked the list id, we could avoid the status field entirely, and have `get_status` calculate the actual value based on the list id?

There's no flaws I can see with the current logic, but there's a risk future devs could accidentally end up getting these two properties out of sync